www/src/public/js/controllers/builder/main.js

```diff
@@ -2070,6 +2308,332 @@
+                        break;
+                    case "P":
+                        //2 hpr, 1 mvr
+                        runeStats.rent += 450;
+                        runeStats.hpr += 2;
+                        runeStats.mvr++;
+                        runeStats.charmName = runeStats.charmName + "Kenaz/";
+                        break;
+                    case "Q":
+                        //2 bonus accuracy
+                        runeStats.rent += 360;
+                        runeStats.rangedAccuracy += 2;
+                        runeStats.charmName = runeStats.charmName + "Perthro/";
+                        break;
+                    case "R":
+                        //-3 ac
+                        runeStats.rent += 135;
+                        runeStats.ac -= 3;
+                        runeStats.charmName = runeStats.charmName + "Thurisaz/";
+                        break;
+                    case "S":
+                        //2 spell crit
+                        runeStats.rent += 103;
+                        runeStats.spellcrit += 2;
+                        runeStats.charmName = runeStats.charmName + "Hagalaz/";
+                        break;
+                    case "T":
+                        //2 spell dam
+                        runeStats.rent += 675;
+                        runeStats.spelldam += 2;
+                        runeStats.charmName = runeStats.charmName + "Nauthiz/";
+                        break;
+                    case "U":
+                        //2 mana redux
+                        runeStats.rent += 203;
+                        runeStats.concentration++;
+                        runeStats.manaReduction += 2;
+                        runeStats.charmName = runeStats.charmName + "Sowilo/";
+                        break;
+                    case "V":
+                        //detect invis
+                        runeStats.rent += 900;
+                        runeStats.charmName = runeStats.charmName + "Jera/";
+                        break;
+                    case "W":
+                        //see dark
+                        runeStats.rent += 900;
+                        runeStats.charmName = runeStats.charmName + "Dagaz/";
+                        break;
+                    case "X":
+                        //detect illusion
+                        runeStats.rent += 900;
+                        runeStats.charmName = runeStats.charmName + "Othala/";
+                        break;
+                    case "Y":
+                        //sneak
+                        runeStats.charmName = runeStats.charmName + "Ingwaz/";
+                        break;
+                }
+            }
+
+            return runeStats;
+        };
+        
+        /** swaps isRuneCrafting for modal hiding purposes **/
+        $scope.runeCraftButtonClick = function() {
+               $scope.isRuneCrafting = !$scope.isRuneCrafting;
+               console.log("Button click");
```

Comment:
Is this log still necessary?

Reply:
nope. I was trying to understand why things were working with this. and not $scope.